identity/src/main/java/com/android/identity/GattServer.java

```diff
@@ -184,7 +187,7 @@ public void onConnectionStateChange(BluetoothDevice device, int status, int newS
             mCurrentConnection = device;
 
         } else if (newState == BluetoothProfile.STATE_DISCONNECTED) {
-            if (!device.getAddress().equals(mCurrentConnection.getAddress())) {
+            if (device != null && !device.getAddress().equals(mCurrentConnection.getAddress())) {
```

Comment:
Good fix, but probably shouldn't be in this PR as it appears unrelated.

Reply:
This was in response to a crash I was getting, I'll do it proper and create a new issue / patch for this